gtsfm/utils/images.py

```diff
@@ -118,6 +118,45 @@ def resize_image(image: Image, new_height: int, new_width: int) -> Image:
     return Image(resized_value_array)
 
 
+def get_downsample_factor_per_axis(img: Image, max_resolution: int) -> Tuple[float, float, int, int]:
```

Comment:
Nit: maybe this function should accept only the dimensions of the image?

Reply:
I was thinking about that, but the way I've implemented it, we need to feed a dummy image through the resize function to see how OpenCV will round the dims. So rather than make a dummy image, I figured a single argument would give it to us, along with the dimensions.